architecture/core.rst

```diff
@@ -390,6 +390,183 @@ normal world supports scheduling of processes. Even on UP systems, supporting
+Interrupt management resources are declared in header file
+kernel/interrupt.h_. Main API function are:
+
+    - ``interrupt_enable()`` and ``interrupt_disable()`` to respectively
+      enable and disable and interrupt.
+
+    - ``interrupt_mask()`` and ``interrupt_unmask()`` to respectively
+      mask/unmask and interrupt.
+
+    - ``interrupt_configure()`` to configure an interrupt detection mode
+      and the interrupt priority.
+
+    - ``interrupt_add_handler()`` and ``interrupt_remove_handler()`` to
+      respectively register and unregister an interrupt handler function
+      for a given interrupt.
+
+When configuration switch ``CFG_DRIVERS_INTERRUPT`` is enabled, core
+offers a means to use Device Tree to describe interrupts topologyin the
+platform. When used, the platform must provide a secure devicetree blob
+(DTB) which platform and device drivers can use to getreferences to
+interrupt controllers and relative interrupt numbers theyconsume.
+Main API functions are:
+
+    - ``dt_register_interrupt_provider()`` to register an interrupt
+      controller related to a given node in the Device Tree. This function
+      requires, among its arguments, a callback function that is called when
+      an interrupt consumer requires a reference to a specific interrupt
+      exposed by the controller.
+
+    - ``dt_get_interrupt()``, ``dt_get_interrupt_by_index()`` and
+      ``dt_get_interrupt_by_name()`` for an interrupt consumer to get the
+      reference to the interrupt controller and number it consumes, based
+      on the properties found in the consumer Device Tree node.
+
+**Interrupt controller drivers**
+
+Interrupt controller instance (``struct itr_chip``) registers operation
+function handlers for management of the interrupt. An interrupt controller
+must provide operation handler functions ``add``, ``enable`` and ``disable``.
+There are other operation handler functions but these are not mandatory, as
+``mask``, ``unmask``, ``rasie_pi``, ``raise_sgi`` and ``set_priority``.
+
+Operation handler function ``add`` is called when an interrupt consumer
+registers a function handler for a given interrupt.
+
+A reference to CPU root interrupt controller, e.g. a GIC instance on Arm
+architecture CPUs, can be retrieved with API function ``itr_core_get()``.
+
+**Interrrupt handlers**
+
+Interrupt handler functions are functions registered by drivers that
+core shall call when the related interrupt occurs. Structure struct itr_handler
+references a handler. It contains the handler function entry point as well
+as interrupt information: its number, its interrupt controller device and
+a few fields.
+
+An interrupt handler function return value is of type ``enum itr_return``.
+It shall return ``ITRR_HANDLED`` when the interrupt is served and
+``ITRR_NONE`` when the interrupt cannot be served.
+
+The interrupt handler executes in an interrupt context, not in a thread
+context. The handler must therefore execute very quickly. If there are
```

Comment:
"as quickly as possible"?
Maybe say why it is so (what happens when a handler runs for a long time)